tensorflow/core/grappler/optimizers/generic_layout_optimizer.cc

```diff
@@ -414,6 +421,15 @@ Status GenericLayoutOptimizer::Optimize(Cluster* cluster,
   const bool is_aggressive = opt_level_ == RewriterConfig::AGGRESSIVE;
 
   TransposeContext context;
+  if (!enforced_layout_.empty()
+      && enforced_layout_ != "NHWC"
+      && enforced_layout_ != "NCHW") {
+    return Status(tensorflow::error::Code::INVALID_ARGUMENT,
+                    absl::StrCat("Invalid value for enforced_layout: ", enforced_layout_,
+                    ". Supported layouts: 'NHWC', 'NCHW'."));
+  }
```

Comment:
Would it be better to move this to the constructor of the optimizer?

Reply:
I agree

Reply:
Other optimizers do the option checking in the `Optimizer` function (for better error status handling)
Hence, I believe that we should leave it in this function... I move it to the beginning of the function